components/psst/psst_service.cc

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/psst/psst_service.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "base/containers/contains.h"
+#include "base/containers/flat_set.h"
+#include "base/files/file_path.h"
+#include "base/functional/callback_forward.h"
+#include "base/logging.h"
+#include "base/memory/singleton.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/brave_component_updater/browser/dat_file_util.h"
+#include "brave/components/psst/psst_rule.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+#include "url/gurl.h"
+#include "url/origin.h"
+
+namespace psst {
+
+constexpr char kJsonFile[] = "psst.json";
+constexpr char kScriptsDir[] = "scripts";
+
+using base::AutoLock;
+
+// static
+PsstService* PsstService::GetInstance() {
+  return base::Singleton<PsstService>::get();
+}
+
+PsstService::PsstService() = default;
```

Comment:
calling this `PsstService` is not great imo because it's overloading the term `Service` even more than it already is. Maybe `PsstRuleMatcher` or something?

Reply:
I still think the Service is useful for someone reading the code and understanding what's happening, I'm guessing this was a nit, if not let me know. I added Rule to the name.

Reply:
I actually pretty strongly disagree here. We are misusing `Service` all over the place and it's creating even more confusion than there already is between keyed services and things in `services` (which are usually mojo services). Calling just any class a service just adds to that confusion.

Reply:
Is there some future functionality for this beyond just matching rules? If so then maybe something else makes sense here